control-plane/pkg/reconciler/base/reconciler.go

```diff
@@ -308,3 +310,22 @@ func (r *Reconciler) OnDeleteObserver(obj interface{}) {
 		r.SecretTracker.OnDeletedObserver(obj)
 	}
 }
+
+func (r *Reconciler) DeleteResource(ctx context.Context, logger *zap.Logger, uuid types.UID, ct *contract.Contract, contractConfigMap *corev1.ConfigMap) error {
+	kafkaSourceIndex := coreconfig.FindResource(ct, uuid)
```

Comment:
not sure `kafkaSource` is correct here? 

I see this being invoked via others too :man_shrugging: